esp-hal/src/i2c/master/mod.rs

```diff
@@ -1115,22 +1106,7 @@ impl embedded_hal_async::i2c::I2c for I2c<'_, Async> {
 }
 
 fn async_handler(info: &Info, state: &State) {
-    let regs = info.regs();
-    regs.int_ena().modify(|_, w| {
-        w.end_detect().clear_bit();
-        w.trans_complete().clear_bit();
-        w.arbitration_lost().clear_bit();
-        w.time_out().clear_bit();
-        #[cfg(not(esp32))]
-        w.scl_main_st_to().clear_bit();
-        #[cfg(not(esp32))]
-        w.scl_st_to().clear_bit();
-
-        #[cfg(not(any(esp32, esp32s2)))]
-        w.txfifo_wm().clear_bit();
-
-        w.nack().clear_bit()
-    });
+    info.regs().int_ena().write(|w| unsafe { w.bits(0) });
```

Comment:
in theory a functional change but most probably this matches the original intention